servers/src/main/java/tachyon/master/block/BlockMaster.java

```diff
@@ -278,13 +278,17 @@ public void removeBlocks(List<Long> blockIds) {
           if (masterBlockInfo == null) {
             continue;
           }
+          ArrayList<Long> workersToRemove = new ArrayList<Long>();
           for (long workerId : masterBlockInfo.getWorkers()) {
-            masterBlockInfo.removeWorker(workerId);
+            workersToRemove.add(workerId);
```

Comment:
I don't have strong preference, but the proposed one is simpler :)
But I agree it doesn't matter much here, if it'll be changed soon.